src/material-table.js

```diff
@@ -184,7 +199,7 @@ export default class MaterialTable extends React.Component {
 
     if (propsChanged) {
       const props = this.getProps(this.props);
-      this.setDataManagerFields(props, false, this.props.columns);
```

Comment:
I am not sure if this PR and especially this line would not result in unintended side effects.

Reply:
Hey all,

Thanks for taking a look at this.

Yea, I've gone ahead and checked references to this parameter in `setDataManagerFields`. Other than the usage that I've added, it looks to be passed to `DataManager.setColumns`, where it's used to "keep tabledata from previous column".

That being said, I don't see why `this.props.columns` was being passed here in the first place? Surely `prevProps.columns` was meant to be passed here all along? I noticed that `props.columns` is never actually modified by the above `this.getProps(this.props)`. So `props.columns` should be unmodified compared to `this.props.columns`.

I also noticed here: https://github.com/material-table-core/core/blob/master/src/utils/data-manager.js#L119. That prevColumns is only ever undefined. `Id` doesn't seem to exist on the column. Only in `column.tableData` (I checked through debugger). Not sure if this is a bug? I couldn't find documentation of an `Id` prop anywhere for columns.

In any case, I haven't found any possible side effects so far. Unless I am misunderstanding the usage of some of these variables?